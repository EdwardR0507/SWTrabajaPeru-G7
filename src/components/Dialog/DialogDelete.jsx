import { React, useState } from "react";
import {
  makeStyles,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
} from "@material-ui/core/";
import DeleteIcon from "@material-ui/icons/Delete";
import theme from "../../themes/themes";

const useStyles = makeStyles(() => ({
  button: {
    width: "9em",
    height: "2.8em",
    fontSize: "0.9em",
    lineHeight: "16px",
    letterSpacing: "1.25px",
  },
  cancelButton: {
    background: theme.palette.secondary.main,
    color: theme.colorLetter.primary.main,
    "&:hover": {
      background: theme.palette.secondary.main,
    },
  },
}));

const DialogDelete = ({ handleDelete, cat_nombre, cat_id }) => {
  const classes = useStyles();
  const [open, setOpen] = useState(false);

  const handleClickOpen = () => {
    setOpen(true);
  };

  const handleClose = () => {
    setOpen(false);
  };

  return (
    <>
      <Button
        variant="contained"
        endIcon={<DeleteIcon />}
        role="open-button"
        className={classes.button}
        onClick={handleClickOpen}
      >
        Eliminar
      </Button>
      <Dialog
        open={open}
        disableScrollLock
        role="close-button"
        onClose={handleClose}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <DialogTitle id="alert-dialog-title">
          {`¿Está seguro que desea eliminar el servicio de ${cat_nombre}?`}
        </DialogTitle>
        <DialogContent>
          <DialogContentText id="alert-dialog-description">
            Al eliminar el servicio de {cat_nombre} será inaccesible toda la
            información relacionada con este servicio. Para restablecer el
            servicio deberá agregarlo nuevamente
          </DialogContentText>
        </DialogContent>
        <DialogActions>
          <Button
            onClick={() => handleDelete(cat_id)}
            variant="contained"
            color="primary"
          >
            Aceptar
          </Button>
          <Button
            onClick={handleClose}
            className={classes.cancelButton}
            autoFocus
          >
            Cancelar
          </Button>
        </DialogActions>
      </Dialog>
    </>
  );
};
export default DialogDelete;
