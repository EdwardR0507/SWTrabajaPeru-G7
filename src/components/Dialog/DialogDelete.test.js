/*Importamos las librerias principales*/
import React from "react";
import { shallow } from "enzyme";
import DialogDelete from "./DialogDelete"

describe(("<DialogDelete />"), () => {
    it("render DialogLogin", () => {
        const wrapper = shallow(<DialogDelete/>);
        expect(wrapper).toMatchSnapshot()
    })
    it('handleClose test', () => {
        const wrapper = shallow(<DialogDelete />);
        const button = wrapper.find({ role: "open-button" });
        button.simulate("click");
        expect(wrapper.state().open).toBe(true)
      })
      it('handleOpen test', () => {
        const wrapper = shallow(<DialogDelete />);
        const button = wrapper.find({ role: "close-button" });
        button.simulate("click");
        expect(wrapper.state().open).toBe(false)
      })
}) 