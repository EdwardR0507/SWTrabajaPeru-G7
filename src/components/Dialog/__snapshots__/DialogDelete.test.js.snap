// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DialogDelete /> render DialogLogin 1`] = `ShallowWrapper {}`;
